scale-out-relay/src/main/scala/com/normation/plugins/scaleoutrelay/ScaleOutRelayService.scala

```diff
@@ -57,28 +65,110 @@ object ScaleOutRelayService {
+          _ <- woLDAPNodeRepository.deleteNode(nodeInf.node, modId, actor, reason).chainError(s"Rollback promote to relay failed : removed ${nodeInf.node} failed")
+          _ <- woLDAPNodeGroupRepository.deletePolicyServerTarget(ruleTarget).chainError(s"Rollback promote to relay failed : removed rule target ${ruleTarget.target} failed")
+          _ <- woLDAPNodeGroupRepository.delete(nodeGroup.id, modId, actor, Some("Promote node to relay rollback")).chainError(s"Rollback promote to relay failed : removed node group ${nodeGroup.id.value} failed")
+          _ <- woDirectiveRepository.deleteActiveTechnique(activeTechniqueId, modId, actor, Some("Promote node to relay rollback")).chainError(s"Rollback promote to relay failed : removed active technique ${activeTechniqueId.value} failed")
+          _ <- woDirectiveRepository.deleteActiveTechnique(activeTechniqueIdCommon, modId, actor, Some("Promote node to relay rollback")).chainError(s"Rollback promote to relay failed : removed active technique ${activeTechniqueIdCommon.value} failed")
+          _ <- woRuleRepository.delete(ruleSetup.id, modId, actor, Some("Promote node to relay rollback")).chainError(s"Rollback promote to relay failed : removed rule  ${ruleSetup.id.value} failed")
+        } yield ()
+      }
+      _ <- woRuleRepository.create(ruleDistribPolicy, modId, actor, reason).catchAll { err =>
+        val e = Chained(s"Trying to remove residual object : rule ${ruleDistribPolicy.id.value}", err)
+        PluginLogger.error(e)
+        for {
+          _ <- woLDAPNodeRepository.deleteNode(nodeInf.node, modId, actor, reason).chainError(s"Rollback promote to relay failed : removed ${nodeInf.node} failed")
+          _ <- woLDAPNodeGroupRepository.deletePolicyServerTarget(ruleTarget).chainError(s"Rollback promote to relay failed : removed rule target ${ruleTarget.target} failed")
+          _ <- woLDAPNodeGroupRepository.delete(nodeGroup.id, modId, actor, Some("Promote node to relay rollback")).chainError(s"Rollback promote to relay failed : removed node group ${nodeGroup.id.value} failed")
+          _ <- woDirectiveRepository.deleteActiveTechnique(activeTechniqueId, modId, actor, Some("Promote node to relay rollback")).chainError(s"Rollback promote to relay failed : removed active technique ${activeTechniqueId.value} failed")
+          _ <- woDirectiveRepository.deleteActiveTechnique(activeTechniqueIdCommon, modId, actor, Some("Promote node to relay rollback")).chainError(s"Rollback promote to relay failed : removed active technique ${activeTechniqueIdCommon.value} failed")
+          _ <- woRuleRepository.delete(ruleSetup.id, modId, actor, Some("Promote node to relay rollback")).chainError(s"Rollback promote to relay failed : removed rule  ${ruleSetup.id.value} failed")
+          _ <- woRuleRepository.delete(ruleDistribPolicy.id, modId, actor, Some("Promote node to relay rollback")).chainError(s"Rollback promote to relay failed : removed rule  ${ruleDistribPolicy.id.value} failed")
```

Comment:
You should have rollbackfunction, one of each step that call the previous step 

def rollbackDistributePolicy (nodeId, modId, actor) = {
  for {
  _ <- woRuleRepository.delete(ruleDistribPolicyId(nodeId), modId, actor, Some("Rollback of distribute policy creation of policy server ${nodeId.value}")).chainError(s"Removing rule {ruleId} failed")
  _ <- rollbackCommonRule(nodeId, modId,actor)
  }
}


def rollbackCommonRule (nodeId, modId, actor) = {
  for {
  _ <- woRuleRepository.delete(CommonRuleId(nodeId), modId, actor, Some("Rollback of common rule creation of policy server ${nodeId.value}")).chainError(s"Removing rule {ruleId} failed")
  _ <- rollbackCommonRule(nodeId, modId,actor)
  }
}


....

RemoveNodeGroup(nodeId, modId, actor) = { 
  removeNodeGroup
  call restoreRelayToNode
}

restoreRelayToNode {
  deleteRelay
}


```